enhancements/dns/dns-operator-operand-logging-level.md

```diff
@@ -0,0 +1,239 @@
+// +optional
+// +kubebuilder:default=Normal
+LogLevel DNSLogLevel `json:"logLevel,omitempty"`
+```
+
+Both of these new APIs would use the exisiting `DNSLogLevel` definitions:
+
+```go
+
+// +kubebuilder:validation:Enum:=Normal;Debug;Trace
+type DNSLogLevel string
+
+var (
+// Normal is the default.  Normal, working log information, everything is fine, but helpful notices for auditing or common operations.  In kube, this is probably glog=2.
+DNSLogLevelNormal DNSLogLevel = "Normal"
+
+// Debug is used when something went wrong.  Even common operations may be logged, and less helpful but more quantity of notices.  In kube, this is probably glog=4.
+DNSLogLevelDebug DNSLogLevel = "Debug"
+
+// Trace is used when something went really badly and even more verbose logs are needed.  Logging every function call as part of a common operation, to tracing execution of a query.  In kube, this is probably glog=6.
+DNSLogLevelTrace DNSLogLevel = "Trace"
+)
+
+```
+
+### User Stories
+
+* As an OpenShift Cluster Administrator, I want to be able to raise the logging level of the DNS Operator and CoreDNS so that I can more quickly
+track down OpenShift DNS issues.
+
+* Some users recently added the new alert 'CoreDNS is returning SERVFAIL for X% of requests alert' to the recent updates of OCP.
+  Adding this prometheus alert is nice, but it would be more useful we can see which request are getting SERVFAIL response.
+  So we would to enable the log plugin for CoreDNS to log queries.
+  
+* Some user want to avoid use of tcpdump to see the queries and want log plugin to be enabled to log queries in coredns.
+
+* Some users actually want logs of every DNS query for auditing purposes, similar to access logs for ingress or audit logs for the API.
+
+
+### Implementation Details/Notes/Constraints [optional]
+
+
+### Risks and Mitigations
+
+Raising the logging verbosity for any component typically results in larger log files that grow quickly.
+
+
+## Design Details
+
+### Open Questions [optional]
+N/A
+
+### Test Plan
+
+Unit tests will be written to test if setting LogLevel sets the respective logging in CoreDNS.
+Unit tests will be written to test if setting operatorLogLevel sets the respective logging in DNS Operator.
+
+### Graduation Criteria
+
+N/A
+
+#### Dev Preview -> Tech Preview
+
+N/A
+
+#### Tech Preview -> GA
+
+N/A
+
+#### Removing a deprecated feature
+
+N/A
+
+### Upgrade / Downgrade Strategy
+
+On downgrade, any logging options are ignored by the DNS Operator and CoreDNS.
+The downgraded operator will update the configmap and delete the log stanzas.
+
+
+### Version Skew Strategy
+
+N/A
+
+## Implementation History
+
+[API Implementation](https://github.com/openshift/api/pull/1031/commits/b606c856dae1b68383f0d068f78ca0823c82ad03/)
+
+## Drawbacks
+
+
+## Alternatives
+
+* Don't provide any DNS logging level APIs for the operator and coredns (current behavior)
+* Raise current verbosity of the DNS Operator and coredns (not desirable)
```

Comment:
A third alternative is tcpdump.  

Normally we'd want a little more discussion as to why the alternative was not considered.